scrapy/core/downloader/contextfactory.py

```diff
@@ -17,4 +23,6 @@ def getContext(self, hostname=None, port=None):
         # Enable all workarounds to SSL bugs as documented by
         # http://www.openssl.org/docs/ssl/SSL_CTX_set_options.html
         ctx.set_options(SSL.OP_ALL)
+        if hostname and ClientTLSOptions is not None: # workaround for TLS SNI
+            ClientTLSOptions(hostname, ctx)
```

Comment:
From the docstring and the import line, I just figured this was only applicable to SSL connections anyway.
We're also only accepting TLSv1 connections here (`self.method = SSL.TLSv1_METHOD`).
(Anything else is just stupid now anyway, after SSL3 has mostly left the building).
So there isn't much need to test other protocols for _this_ class ( unless someone overrides the TLS line here, then that'd be an unsupported case ).

That's not to say that a test-case wouldn't be great.
But I noticed that all my SNI sites also work without that fix (strange), so I have no real host to point a testcase at, and it should preferrably be an offline test anyway? If such a test can even be fabricated.
Any help welcome.